### PR TITLE
[storage.blob] Remove aiohttp as dependency for storage.blob.aio

### PR DESCRIPTION
**Pull request by Stevenjin8**
# Description

`aio/_download_async.py` used to import `aiohttp.client_exceptions.ClientPayloadError` to catch it if it was raised by the transport. The client should be agnostic to the transport implementation hence not import this error. Also, the `AioHttpTransportResponse` already catches `aiohttp.client_exceptions.ClientPayloadError` so this exception should never be raised by the transport.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/storage/azure-storage-blob/azure/storage/blob/aio/_download_async.py [hunk 1]
```diff
@@ -111,14 +109,8 @@ async def _download_chunk(self, chunk_start, chunk_end):
                         **self.request_options
                     )
                     retry_active = False
-
                 except HttpResponseError as error:
                     process_storage_error(error)
-                except ClientPayloadError as error:
```
Comment (Stevenjin8) [inline sdk/storage/azure-storage-blob/azure/storage/blob/aio/_download_async.py:117]: `AioHttpTransport` will catch `ClientPayloadError` and raise it as a `ServiceResponseError` which inherits from `HttpsResponseError`.

**Review by jalauzon-msft — APPROVED** [review]
Thanks for fixing this up!

File: sdk/storage/azure-storage-blob/CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,7 @@
 ### Features Added
 
 ### Bugs Fixed
+- Removed `aiohttp` import from `azure.storage.blob._download_async`.
```
Comment (lmazuel) [inline sdk/storage/azure-storage-blob/CHANGELOG.md:8]:
```suggestion
- Removed `aiohttp` import from `azure.storage.blob._download_async`.  #24965
```

File: sdk/storage/azure-storage-blob/azure/storage/blob/aio/_download_async.py [hunk 3]
```diff
@@ -356,6 +357,12 @@ async def _initial_request(self):
                     self.size = self._file_size
                 retry_active = False
 
+            except IncompleteReadError as error:
+                retry_total -= 1
+                if retry_total <= 0:
+                    raise ServiceResponseError(error, error=error)
+                await asyncio.sleep(1)
```
Comment (johanste) [inline sdk/storage/azure-storage-blob/azure/storage/blob/aio/_download_async.py:364]: Not a new issue, but why aren't we honoring the retry timeouts?

**Review by johanste — APPROVED** [review]

File: sdk/storage/azure-storage-blob/CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,7 @@
 ### Features Added
 
 ### Bugs Fixed
+- Removed `aiohttp` import from `azure.storage.blob.aio._download_async`.  #24965
```
Comment (vincenttran-msft) [inline sdk/storage/azure-storage-blob/CHANGELOG.md:8]: Super nit: See line 69 for an example of mentioning an issue in the changelog (just for consistency sake 😄)

**Review by vincenttran-msft — APPROVED** [review]
LGTM! Thanks for catching this and also promptly getting a PR out 😄

**Comment by Stevenjin8** [comment]
I think we should get rid of the retires all together (#25017) since there is a retry policy in the pipeline.

File: sdk/storage/azure-storage-blob/azure/storage/blob/aio/_download_async.py (hunk 3 above)
Comment (lmazuel) [reply sdk/storage/azure-storage-blob/azure/storage/blob/aio/_download_async.py:364]: I'm resolving this discussion, as the scope of Steven was really just the exception, but this is a good question.

**Pull request merged**